keras_cv/models/darknet.py

```diff
@@ -201,15 +202,14 @@ def DarkNet(
 
     if include_top:
         x = layers.GlobalAveragePooling2D(name="avg_pool")(x)
-        x = layers.Dense(
-            classes, activation=classifier_activation, name="predictions"
-        )(x)
-    elif pooling == "avg":
-        x = layers.GlobalAveragePooling2D(name="avg_pool")(x)
-    elif pooling == "max":
-        x = layers.GlobalMaxPooling2D(name="max_pool")(x)
+        x = layers.Dense(classes, activation=classifier_activation, name="predictions")(x)
+    else:   
+        if pooling == "avg":
+            x = layers.GlobalAveragePooling2D(name="avg_pool")(x)
+        elif pooling == "max":
+            x = layers.GlobalMaxPooling2D(name="max_pool")(x)
 
-    model = keras.Model(inputs, x, name=name, **kwargs)
+    model = tf.keras.Model(inputs, x, name=name, **kwargs)
```

Comment:
This is not a functional subclass model, because `DarkNet` is still a function (not a class). Please take a look at the changes to VGG19 in #1471 as an example.